packages/form-builder/src/blocks/fields/amount/inspector/index.js

```diff
@@ -0,0 +1,73 @@
+                        display: 'flex',
+                        flexDirection: 'column',
+                        gap: '16px',
+                    }}>
+                        {
+                            props.attributes.levels.map((label, index) => {
+                                return (
+                                    <li key={'level-option-inspector-' + index} style={{
+                                        display: 'flex',
+                                        gap: '16px',
+                                        justifyContent: 'space-between',
+                                        alignItems: 'center',
+                                    }}>
+                                        <div style={{position: 'relative'}}>
+                                            <div
+                                                style={{
+                                                    position: 'absolute',
+                                                    height: '100%',
+                                                    display: 'flex',
+                                                    alignItems: 'center',
+                                                    paddingLeft: '13px',
+                                                    fontSize: '.8em',
+                                                }}>
+                                                <spann style={{marginBottom: '1px'}}>$</spann>
+                                            </div>
+                                            <input
+                                                style={{
+                                                    width: '140px',
+                                                    padding: '6px 0 6px 30px',
+                                                }}
+                                                value={label}
+                                                onChange={(val) => {
+                                                    const levels = [...props.attributes.levels];
+                                                    levels[index] = val;
+                                                    props.setAttributes({levels: levels});
+                                                }}
+                                            />
+                                        </div>
+                                        <DeleteButton onClick={() => {
+                                            props.attributes.levels.splice(index, 1);
+                                            props.setAttributes({levels: props.attributes.levels.slice()});
+                                        }} />
+                                    </li>
+                                );
+                            })
+                        }
+                    </ul>
+                )}
+                <AddButton onClick={() => {
+                    const levels = [...props.attributes.levels];
```

Comment:
FYI 
`const levels = [...props.attributes.levels];` 
is the same thing as 
`const levels = props.attributes.levels;`

Reply:
I'm cloning the object here, which seems to be required when passing it to `setAttributes()`.

Reply:
Sorry, I was just pointing out that spreading a single array inside an empty array is the same as referencing the array in the first place, so you don't need to do the spreading here.

Reply:
It was breaking if I don't spread the array into a new array. I think there is an issue with mutability at play. If I update the attribute value directly, passing it to `setAttributes()` does not update the attribute. However, if I spread the value into a new object then it works.

```js
const a = []
const b = a // assign by reference
const c = [ ...a ] // assign by value

console.log( a === b ) // true
console.log( a === c ) // false
```

Reply:
Just because I know y'all would care, technically _everything_ in JavaScript is passed by value, including objects. Objects, however, are a set of references — a reference of references. So when passing an object, it creates a copy of reference. This means that the underlying references are the same, which gives it the feeling of being by reference. This evaluation strategy is called [Call by sharing](https://en.wikipedia.org/wiki/Evaluation_strategy#Call_by_sharing).

So when you use `Object.assign()` or spreading, the underlying object references (keep in mind that arrays are objects in JS) are resolved to pass by value. BUT! This is only true for top-level values. Call by sharing takes effect again for nested objects:

```js
let foo = { a: 1, b: 2, c: { d: 3 } };
let bar = { ...foo };

foo.a = 10;
foo.c.e = 4;

bar.a; // 1
bar.c.e; // 4
```